cluster/ubuntu/util.sh

```diff
@@ -358,13 +358,13 @@ function provision-master() {
                             create-kube-apiserver-opts "${SERVICE_CLUSTER_IP_RANGE}" "${ADMISSION_CONTROL}" "${SERVICE_NODE_PORT_RANGE}"; \
                             create-kube-controller-manager-opts "${NODE_IPS}"; \
                             create-kube-scheduler-opts; \
-                            create-flanneld-opts "127.0.0.1"; \
                             sudo -p '[sudo] password to start master: ' cp ~/kube/default/* /etc/default/ && sudo cp ~/kube/init_conf/* /etc/init/ && sudo cp ~/kube/init_scripts/* /etc/init.d/ ;\
                             sudo groupadd -f -r kube-cert; \
                             ${PROXY_SETTING} sudo -E ~/kube/make-ca-cert.sh ${MASTER_IP} IP:${MASTER_IP},IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.cluster.local; \
                             sudo mkdir -p /opt/bin/ && sudo cp ~/kube/master/* /opt/bin/; \
                             sudo service etcd start; \
-                            sudo FLANNEL_NET=${FLANNEL_NET} -b ~/kube/reconfDocker.sh "a";"
+                            sudo FLANNEL_NET=${FLANNEL_NET} ~/kube/reconfDocker.sh "a"; \
+                            create-flanneld-opts "127.0.0.1";"
```

Comment:
flanneld is definitely up from my env. Where did you catch this? 


Friday, October 2, 2015, 16:46 +0800 from thaerlo  <notifications@github.com>:
>In  cluster/ubuntu/util.sh :
>>                              sudo -p '[sudo] password to start master: ' cp ~/kube/default/* /etc/default/ && sudo cp ~/kube/init_conf/* /etc/init/ && sudo cp ~/kube/init_scripts/* /etc/init.d/ ;\
>                              sudo groupadd -f -r kube-cert; \
>                              ${PROXY_SETTING} sudo -E ~/kube/make-ca-cert.sh ${MASTER_IP} IP:${MASTER_IP},IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.cluster.local; \
>                              sudo mkdir -p /opt/bin/ && sudo cp ~/kube/master/* /opt/bin/; \
>                              sudo service etcd start; \
> -                            sudo FLANNEL_NET=${FLANNEL_NET} -b ~/kube/reconfDocker.sh "a";"
> +                            sudo FLANNEL_NET=${FLANNEL_NET} ~/kube/reconfDocker.sh "a"; \
> +                            create-flanneld-opts "127.0.0.1";"
>Executing reconfDocker "ai" will execute:
>source /run/flannel/subnet.env
>echo DOCKER_OPTS=\"${DOCKER_OPTS} -H tcp://127.0.0.1:4243 -H unix:///var/run/docker.sock \
>--bip=${FLANNEL_SUBNET} --mtu=${FLANNEL_MTU}\" > /etc/default/docker
>If the flannel interface is not up this will fail and docker wont be configured correctly on the masterandnode node.
>With this change we are guaranteeing that the flannel interface is up and running before setting up the docker interface.
>—
>Reply to this email directly or  view it on GitHub .